skynet/modules/ttt/openai_api/app.py

```diff
@@ -9,37 +12,76 @@
     llama_path,
     openai_api_base_url,
     openai_api_server_port,
-    vllm_server_path,
 )
 from skynet.logs import get_logger
-from skynet.utils import get_device
+from skynet.utils import create_app, dependencies, get_device, responses
 
-proc = None
 use_vllm = get_device() == 'cuda'
 
 log = get_logger(__name__)
 
 
-def initialize():
-    log.info('Starting OpenAI API server...')
+async def run_vllm_server(args, main_app: FastAPI):
+    from vllm.entrypoints.launcher import serve_http
+    from vllm.entrypoints.openai.api_server import (
+        build_async_engine_client,
+        init_app_state,
+        lifespan,
+        router,
+        TIMEOUT_KEEP_ALIVE,
+    )
+
+    async with build_async_engine_client(args) as engine_client:
+        app = create_app(lifespan=lifespan)
+        app.include_router(router, dependencies=dependencies, responses=responses)
+
+        model_config = await engine_client.get_model_config()
+        init_app_state(engine_client, model_config, app.state, args)
+
+        main_app.mount('/openai', app)
+
+        shutdown_task = await serve_http(
```

Comment:
Is this necessary? Since the app is served by our uvicorn, what does this serve_http do? Why do we need to specify port and host and everything?